controllers/druid/druid_controller_test.go

```diff
@@ -188,6 +193,129 @@ var _ = Describe("Druid Operator", func() {
 				}, timeout, interval).Should(BeTrue())
 			})
 		}
+
+		// Test statefulsets replica count and update the replica count then match
+		for _, componentName := range druidComponents {
+			componentName := componentName
+			It(fmt.Sprintf("Test kubernetes service for %s", componentName), func() {
+				createdService := &v1.Service{}
+				serviceName := fmt.Sprintf("druid-%s-%s", druidCR.Name, componentName)
+				serviceNamespacedName := types.NamespacedName{Name: serviceName, Namespace: druidCR.Namespace}
+
+				// Checking  the kubernetes service Type
+				By(fmt.Sprintf("By checking kubernetes service type %s ", serviceName))
+				Eventually(func() bool {
+					err := k8sClient.Get(context.TODO(), serviceNamespacedName, createdService)
+					return err == nil
+				}, timeout, interval).Should(BeTrue())
+				Expect(createdService.Spec.Type).To(Equal(druidCR.Spec.Services[0].Spec.Type))
+				// Kubernetes service check for targetport and port
+				By(fmt.Sprintf("By checking kubernetes service port for %s ", serviceName))
+				Expect(createdService.Spec.Ports[0].Port).To(Equal(druidCR.Spec.Nodes[componentName].DruidPort))
+				Expect(createdService.Spec.Ports[0].TargetPort.IntVal).To(Equal(druidCR.Spec.Nodes[componentName].DruidPort))
+			})
+		}
```

Comment:
FYI: I had one more test in mind to assert the annotations in. a service. To enable cloud loadbalancers, druid CR nodespec.service supports adding annotations, those scenarios can also be asserted.
We can take that in a separate PR also.